dali/pipeline/data/tensor_vector.cc

```diff
@@ -149,8 +149,10 @@ void TensorVector<Backend>::SetSize(int new_size) {
 
 
 template <typename Backend>
-void TensorVector<Backend>::set_type(const TypeInfo &new_type) {
-  type_ = new_type;
+void TensorVector<Backend>::set_type(DALIDataType new_type) {
+  if (type_.id() == new_type)
+    return;
+  type_ = TypeTable::GetTypeInfo(new_type);
   tl_->set_type(new_type);
```

Comment:
Maybe we should keep `inline void set_type(const TypeInfo& new_type) {` and use it here, as you call `GetTypeInfo` anyway you can use it directly instead calling it again inside `set_type`.

Reply:
I specifically want to discourage that.